From 16288a68c6f6a394d51ac1aac91c7cb21c1d115f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:33:44 -0400
Subject: [alsa] hda: drop def association and sequence from pinconf comparing

Message-id: <140972823573-315-patchset-jkysela@redhat.com>
Patchwork-id: 91011
O-Subject: [RHEL 7.1 PATCH 314/442] ALSA: hda - drop def association and sequence from pinconf comparing
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

A lot a machine have the same codec, but they have different default
pinconf setting just because the def association and sequence is
different, as a result they can't share a hda_pintbl[], to overcome
it, we don't compare def association and sequence in the pinconf
matching.

Signed-off-by: Hui Wang <hui.wang@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: c687200b9d4ef60042a50f7d942cfef120cc7bf1)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_auto_parser.c b/sound/pci/hda/hda_auto_parser.c
index b684c6e..3cf9137 100644
--- a/sound/pci/hda/hda_auto_parser.c
+++ b/sound/pci/hda/hda_auto_parser.c
@@ -844,7 +844,8 @@ static bool pin_config_match(struct hda_codec *codec,
 {
  for (; pins->nid; pins++) {
   u32 def_conf = snd_hda_codec_get_pincfg(codec, pins->nid);
-  if (pins->val != def_conf)
+  u32 mask = 0xffffff00;
+  if ((pins->val & mask) != (def_conf & mask))
    return false;
  }
  return true;
-- 
1.7.1